src/main/java/frc/robot/commands/autonomous/FiveCycle.java

```diff
@@ -82,7 +83,9 @@ public FiveCycle(RobotContainer robotContainer, boolean nonProcessorSide) {
 
   public Command autoHelper(PathPlannerPath pathOne, PathPlannerPath pathTwo) {
     return Commands.sequence(
-        Commands.waitSeconds(waitTime),
+        // wait until coral is loaded
+        Commands.waitUntil(() -> robotContainer.endEffector.endEffectorIO.coralLoaded()),
```

Comment:
use a `BooleanSupplier` in the constructor to provide this logic.
That way you don't need to make `endEffector` public, and can centralize the logic into RobotContainer